opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -256,6 +263,9 @@ def end(self, end_time: int = None) -> None:
 
         self.span_processor.on_end(self)
 
+        if self.status is None:
```

Comment:
We must do this before calling the span processor.

Reply:
Fixed!